src/generated/xinput.rs

```diff
@@ -3900,47 +3853,51 @@ impl Serialize for FeedbackCtlDataBell {
 pub enum FeedbackCtlData {
     Keyboard(FeedbackCtlDataKeyboard),
     Pointer(FeedbackCtlDataPointer),
-    String(FeedbackCtlDataString),
+    Keysyms(Vec<xproto::Keysym>),
     IntToDisplay(i32),
     Led(FeedbackCtlDataLed),
     Bell(FeedbackCtlDataBell),
 }
 impl FeedbackCtlData {
     fn try_parse(value: &[u8], class_id: u8) -> Result<(Self, &[u8]), ParseError> {
+        let switch_expr = class_id;
         let mut outer_remaining = value;
         let mut parse_result = None;
-        if class_id == u8::from(FeedbackClass::Keyboard) {
+        if switch_expr == u8::from(FeedbackClass::Keyboard) {
             let (keyboard, new_remaining) = FeedbackCtlDataKeyboard::try_parse(outer_remaining)?;
             outer_remaining = new_remaining;
             assert!(parse_result.is_none(), "The XML should prevent more than one 'if' from matching");
             parse_result = Some(FeedbackCtlData::Keyboard(keyboard));
         }
-        if class_id == u8::from(FeedbackClass::Pointer) {
+        if switch_expr == u8::from(FeedbackClass::Pointer) {
             let (pointer, new_remaining) = FeedbackCtlDataPointer::try_parse(outer_remaining)?;
             outer_remaining = new_remaining;
             assert!(parse_result.is_none(), "The XML should prevent more than one 'if' from matching");
             parse_result = Some(FeedbackCtlData::Pointer(pointer));
         }
-        if class_id == u8::from(FeedbackClass::String) {
-            let (string, new_remaining) = FeedbackCtlDataString::try_parse(outer_remaining)?;
-            outer_remaining = new_remaining;
+        if switch_expr == u8::from(FeedbackClass::String) {
+            let remaining = outer_remaining;
+            let remaining = remaining.get(2..).ok_or(ParseError::ParseError)?;
+            let (num_keysyms, remaining) = u16::try_parse(remaining)?;
+            let (keysyms, remaining) = crate::x11_utils::parse_list::<xproto::Keysym>(remaining, num_keysyms as usize)?;
+            outer_remaining = remaining;
             assert!(parse_result.is_none(), "The XML should prevent more than one 'if' from matching");
-            parse_result = Some(FeedbackCtlData::String(string));
+            parse_result = Some(FeedbackCtlData::Keysyms(keysyms));
```

Comment:
This part is definitely wrong. The relevant XML is:
```xml
            <case name="string">
                <enumref ref="FeedbackClass">String</enumref>
		<pad bytes="2" />
		<field type="CARD16" name="num_keysyms" />
		<list type="KEYSYM"  name="keysyms">
		    <fieldref>num_keysyms</fieldref>
		</list>
            </case>
```

Reply:
Actually... after re-reading this:
- the pad does not contain any useful data
- the length field is not needed since the length of the `Vec` will represent this

Sorry for claiming that this is "obviously wrong". I now think that your new version is correct and a lot better than the old version.